kong/plugins/ai-prompt-decorator/access.lua

```diff
@@ -0,0 +1,42 @@
+local _M = {}
+
+-- imports
+local kong_meta    = require "kong.meta"
+local re_match     = ngx.re.match
+local re_find      = ngx.re.find
+local fmt          = string.format
+local table_insert = table.insert
+--
+
+_M.PRIORITY = 772
+_M.VERSION = kong_meta.version
+
+local function to_chat_prompt(version, role, content)
+  if version == "v1" then
+    return { role = role, content = content }
+  else
+    return nil
+  end
+end
+
+function _M.execute(request, conf) 
+  -- 1. add in-order to the head of the chat
+  if conf.prompts.prepend and #conf.prompts.prepend > 0 then
+    for i, v in ipairs(conf.prompts.prepend) do
+      table.insert(request.messages, i, to_chat_prompt("v1", v.role, v.content))
+    end
+  end
+
+  -- 2. add in-order to the tail of the chat
+  if conf.prompts.append and #conf.prompts.append > 0 then
+    local messages_length = #request.messages
+
+    for i, v in ipairs(conf.prompts.append) do
+      request.messages[i + messages_length] = to_chat_prompt("v1", v.role, v.content)
+    end
+  end
+  
+  return nil, nil
+end
```

Comment:
This `to_chat_prompt` function is unnecessary, and the fact that it can return `nil` is going to be problematic when inserting into a table using an explicit index:

```lua
local t = {}
t[1] = "a"
t[2] = "b"
t[3] = nil -- oops
t[4] = "c"
```

Here is a more idiomatic approach:

```lua
local new_tab = require("table.new")
local EMPTY = {}

function _M.execute(request, conf)
  local prepend = conf.prompts.prepend or EMPTY
  local append = conf.prompts.prepend or EMPTY

  if #prepend == 0 and #append == 0 then
    return
  end

  local old_messages = request.messages
  local new_messages = new_tab(#append + #prepend + #old_messages, 0)
  request.messages = new_messages

  local n = 0

  for _, v in ipairs(prepend) do
    n = n + 1
    new_messages[n] = { role = v.role, content = v.content }
  end

  for _, msg in ipairs(request.messages) do
    n = n + 1
    new_messages[n] = msg
  end

  for _, v in ipairs(append) do
    n = n + 1
    new_messages[n] = { role = v.role, content = v.content }
  end
end
```

You can always introduce `to_chat_prompt` later on if you need to introduce new functionality, but better to start with the simpler code now.

You might also consider using [table.clone](https://github.com/openresty/luajit2#tableclone) for `append` and `prepend` if they are simply being copied 1-for-1 and don't contain any nested elements (since it's a shallow clone).